pkg/oc/lib/describe/projectstatus_test.go

```diff
@@ -658,25 +658,32 @@ func TestPrintMarkerSuggestions(t *testing.T) {
 	}
 }
 
-// ReadObjectsFromPath reads objects from the specified file for testing.
-func readObjectsFromPath(path, namespace string, decoder runtime.Decoder, typer runtime.ObjectTyper) ([]runtime.Object, error) {
+// readObjectsFromPath reads objects from the specified file for testing.
+func readObjectsFromPath(path, namespace string) ([]runtime.Object, error) {
 	data, err := ioutil.ReadFile(path)
 	if err != nil {
 		return nil, err
 	}
-	data, err = yaml.ToJSON(data)
-	if err != nil {
-		return nil, err
-	}
+	// Create a scheme with only the types we care about, also to ensure we
+	// are not messing with the bult-in schemes.
+	// We need to perform roundtripping to invoke defaulting, just deserializing
+	// files is not sufficient here.
+	scheme := runtime.NewScheme()
+	kubernetesscheme.AddToScheme(scheme)
+	api.Install(scheme)
+	install.InstallInternalKube(scheme)
+	install.InstallInternalOpenShift(scheme)
+	codecs := serializer.NewCodecFactory(scheme)
+	decoder := codecs.UniversalDecoder()
 	obj, err := runtime.Decode(decoder, data)
```

Comment:
@deads2k I had to leave this decode as it was before instead of deserializer b/c I needed defaulting to kick in.

Reply:
It's fine as a stepping stone.  We'll need to sort it out before we can snip `oc` entirely.